### ISSUE TITLE
Separate out Building levels and Roof Levels quest

### ISSUE THREAD

**Issue by grrrrr** [streetcomplete/StreetComplete#5817]
**Use case**

Some time back it was decided that the two quests building levels and roof levels should be combined.   I am suggesting that these get separated out again as it makes it hard to reliably complete the quest.

- This seems to be the only quest where two separate osm tags are required to complete the quest.
  - `building:levels`
  - `roof:levels`  
- It is quite often impossible to know how many roof levels, if any in a house.  Where as you can very easily see the building levels. 

It also makes the assumption that all buildings must have roof levels but there's no required mapping of  `building:levels:underground`

The is a somewhat similar problem to the quest removed in v59.0-alpha1 - _"Asking "Can you pick up parcels here?" and "Can you drop off parcels here?" **because it turns out that it is often not surveyable** and the answer is always the same for one brand of parcel lockers"_

This requirement, at least in the area I map, has had a detrimental effect on the completion of building mapping.   The building quest has moved from map what you can see to now needing an access to the internal of buildings to ensure the roof levels is correct.   Or in some instances people are putting in junk data for root levels making the map incorrect

The map is no less correct for not having the roof levels tagged but does suffer from not being easily able to map the building level now.

**Proposed Solution**

Revert the task to how it worked in previous years.    Allow building levels to be tagged without the requirement for the separate osm tag roof levels

If they can't be separated out, I would propose:  
- also requiring  `building:levels:underground` along with   - `building:levels` and `roof:levels` to ensure completeness of building mapping in one go as the quest wants
or
-  removing the quest altogether for the same reason as the "Can you pick up parcels here?" quest

**Comment by westnordost** [comment]
Mh no, let's keep it as it is.

The current implementation is the result of several earlier discussions and the arguments you bring up are not new.

Note that this quest is already disabled because it is sometimes hard to answer (amongst other reasons). This reason is shown when users deliberately active the quest.

**Comment by grrrrr** [comment]
Yes,  and look at the previous discussions and you'll see requests (other than mine) to revert it,  the main one against it was yourself, and honestly did not give any really good reasoning why[1].

> Note that this quest is already disabled because it is sometimes hard to answer (amongst other reasons). This reason is shown when users deliberately active the quest.

Because it is disabled by default does not mean forcing people who want to turn it on and _can_ map the building:levels  to either enter buildings

If it is not meant to be turned on,  get rid of it.

Taking GB tag info as an example,  if every building level was to be mapped by Street Complete,  there would be a potential loss of over 1,000,000 buildings with building:levels.  (under 400,000 roof:level tags)

>  it is sometimes hard to answer 

You literally have a "I can't say" option on most if not all quests, it make no sense what you say unless you feel like you can't trust users to accurately map what they see.  

 If I or anyone can't ascertain the building level we can use that,   but now instead of being able to have a street or housing estate with accurate building levels it has to be forgone because mapping we do not have the ability to see in side the root.

1: It is your project,  it is very unfortunate that you are crippling the ability to map such a key tag (building:levels) over the insistence of including a lesser one (roof:levels).

**Comment by matkoniecz** [comment]
> Yes, and look at the previous discussions and you'll see requests (other than mine) to revert it

If you are aware of previous discussions then in such cases it would be helpful to mention it.

It is unlikely that relitigating past issue is going to he helpful. It very very rarely makes sense and in such cases linking previous discussions and discussing why reasons for closing them does not apply anymore (and if situation has no changed and someone is just unhappy about decision then it is even rarer for it to apply)

> it make no sense what you say unless you feel like you can't trust users to accurately map what they see

interface needs to be as clear as possible to trust inexperienced users with reliably answering questions

>  it is very unfortunate that you are crippling the ability to map such a key tag (building:levels) over the insistence of including a lesser one (roof:levels).

the reasoning is NOT about forcing people to map even less important tag, it is too avoid confusion and bad data in case of buildings with nonzero levels in roof

**Comment by rhhsm** [comment]
> * This seems to be the only quest where two separate osm tags are required to complete the quest.

The indoor&outdoor seating quest is similar: indoor seating may not be visible from the street (a basement bar without windows, a restaurant on a higher floor, ...), while outdoor seating may be at the back of a building or not visible in winter. If one is not visible, the quest can't be answered.

**Comment by grrrrr** [comment]
thanks @rhhsm and @mnalis.   It's sad that @westnordost forces how people should map to his preference rather than trusting the user to map and not force errors as a work around.

@matkoniecz 
> If you are aware of previous discussions then in such cases it would be helpful to mention it.

@westnordost brought up the other discussions,  I would think he reference them, I am merely responding to him.

> interface needs to be as clear as possible to trust inexperienced users with reliably answering questions

as @westnordost said it's disabled by default, inexperienced users should not see it (but see below)

> the primary reasoning is (if I remember correctly) NOT about forcing people to map even less important tag, it is too avoid confusion and bad data in case of buildings with nonzero levels in roof

It has literally done the opposite, in areas I map with people putting in random number for roof level.  This I assume is to get building levels to submit.   It's a bit of a pain to have to fix.

So with this option,  we have gone from better mapping of `building:levels` on OSM  to  reduced mapping of `building:levels` or just plain wrong `roof:level`

But it is @westnordost  project,  it's his prerogative to help or hinder the improvement of OSM

**Comment by mnalis** [comment]
**TL;DR:** please @grrrrr  contact those mappers via changeset discussions when you see they are entering incorrect information, and ask why are they doing that, and come back here with links to several of those discussions.  Then we might perhaps have some new information on which to act.

---

> It has literally done the opposite, in areas I map with people putting in random number for roof level. This I assume is to get building levels to submit

Did you @grrrrr  contact those people [on their changesets](https://wiki.openstreetmap.org/wiki/Changeset#Changeset_Discussions) and asked them why are they putting random numbers for roof levels (and could you share links to some of those discussions)? 

If _**any**_ mapper (using _**any**_ editor) is entering _**any**_ incorrect/random data (be it wrong roof numbers or nonexistent POIs or something else) , first step should be to **contact them** and try to understand what they are doing and why. I find it _our duty_  (as more experienced mappers) to do so when we encounter such problems, especially ones made by newbies. 
Be sure to be welcoming to them, but point out the problem. Because, if we don't contact them and explain why what they are doing is a problem for OSM project, they will *surely* continue doing so unknowingly and errors will multiply exponentially over time for many years. And all that incorrect data will be _on our souls_ (not theirs!), as we knew about the problem but ignored acting on it (and they simply didn't know better, and nobody even warned them).

Even if core of the problem actually _is_ confusing editor UI (which is far from proven fact), **still the newbies who made mistakes should first be contacted** as their description why they misunderstood the quest (and why did they misunderstood/ignored the warning text displayed when they tried to enable that quest) and how some quest might make more sense to them would be invaluable information (especially if sample size is more than a single mapper) for even attempting to devise a better solution. 

And in the end, even if it may not be intentional [Vandalism](https://wiki.openstreetmap.org/wiki/Vandalism) but perhaps a confused user, if they fail to respond to repeated attempts to contact them and continue entering incorrect data into OSM, further steps might need to be taken (which are often the same as ones for vandalism - reverting incorrect changesets, contacting [DWG](https://wiki.openstreetmap.org/wiki/Data_Working_Group) for temporary blocks until they read their messages, etc.)
